### DISCUSSION TITLE
Can not increase Charge / Discharge Limit & no Cell informations

### DISCUSSION THREAD

**Discussion by DIVERSDIVERS** [Louisvdw/dbus-serialbattery#509] [Q&A] [answered]
Hello,

Maybe someone can help me. Unfortunately I can't adjust the charge and discharge limit although I changed it in the file, 
see attachment.

Another problem is that I can't see the single cells, what do I have to do?

Thank you

![01](https://user-images.githubusercontent.com/129951689/230057622-f30a7a84-97d8-4f73-a42e-1015679e4515.JPG)
![02](https://user-images.githubusercontent.com/129951689/230057627-771662bf-71b4-4588-bafb-b12a3444312b.JPG)
![03](https://user-images.githubusercontent.com/129951689/230057629-843ff165-8242-458a-bfed-d70ac2c1c4d9.JPG)

**Comment by mr-manuel** [comment, marked as answer]
Hello,

1. What for a BMS are you using?
2. How is the BMS connected?
3. Have you a 8 cell configuration?
4. Which driver version are you using?
5. Which Venus OS Version are you using?
6. Please post the whole configuration.
7. What is the current SoC?

**Reply by DIVERSDIVERS** [reply]
Hi,

1. Daly BMS 125/250 8S
2. via RS485 to USB
3. Yes 8 cell
4. v 0.14.3
5. Color GX Venus OS v 2.93
7. @ the moment 100%

Thanks

**Reply by mr-manuel** [reply]
6. is missing (full config)

**Reply by DIVERSDIVERS** [reply]
Hi.

does this help u or what do you need else?

THX
![Full Solar](https://user-images.githubusercontent.com/129951689/230149469-6bef0f30-2bb6-4feb-863a-3184a04a2bf7.JPG)

**Reply by mr-manuel** [reply]
No, I need the full `utils.py`.

**Reply by DIVERSDIVERS** [reply]
ahhh ok

**Reply by mr-manuel** [reply]
Can you mark this as answer?

**Comment by DIVERSDIVERS** [comment]
here
[utils.zip](https://github.com/Louisvdw/dbus-serialbattery/files/11161929/utils.zip)

**Reply by mr-manuel** [reply]
You have enabled CCCM_CV and CCCM_T. You have to disable it or change the maximum values (60) as well.

https://github.com/Louisvdw/dbus-serialbattery/blob/3685d15a5d1ba801867bbdb98f57fd805ea2c24a/etc/dbus-serialbattery/utils.py#L46-L73

**Reply by DIVERSDIVERS** [reply]
Good morning,

it worked pefect, thx now i understand it better.

Last 2 questions:
- what is in my command wrong to update and it usefull to use last version
- is it possible to get any single cell information

Thx in advance
![001](https://user-images.githubusercontent.com/129951689/230306865-ed378ffd-9674-4120-9514-3f227e688b1d.JPG)
![002](https://user-images.githubusercontent.com/129951689/230306868-2538a6c0-0d7f-476d-9753-94bf0c16f3f0.JPG)

**Reply by mr-manuel** [reply]
Hi,
1. wget is a command to download something from a URL. Before https you inserted some text that is not needed. The last official version is actually 0.14.3. The most up to date version is in the master branch, but it may contain some errors.
Here are some commands on how to install it, if you want to try it out https://github.com/Louisvdw/dbus-serialbattery/issues/421#issuecomment-1465329297
3. Sure, there should be a "Cell Voltages" page in the same list as "Parameters".

**Reply by DIVERSDIVERS** [reply]
hi again :-),

ok than i will use 0.14.3. I made u a donation via PayPal for ur help :-). 
Sry i can find information about every Cell, do i need to change something?

Thx
![001](https://user-images.githubusercontent.com/129951689/230313686-f98f7e95-2c2f-4afc-aba1-822953ab655d.JPG)
![002](https://user-images.githubusercontent.com/129951689/230313689-f3fe4341-28f5-4817-b79d-c8c831a2608c.JPG)

**Reply by mr-manuel** [reply]
Thank you very much!

It seems that the qml files weren't installed.

Try to run `sh /data/etc/dbus-serialbattery/reinstalllocal.sh`.

**Reply by DIVERSDIVERS** [reply]
Ofc you helped me a lot :-)

no effect, do I have to set something else in Venus or is that not possible with a USB to RS485 cable

**Reply by mr-manuel** [reply]
Can you check with the `dbus-spy` command, if you see cell voltages on the dbus?
You have already rebooted the system, right?

**Reply by DIVERSDIVERS** [reply]
ofc rebooted

here the info
![Bild_2023-04-06_110115712](https://user-images.githubusercontent.com/129951689/230328539-2917bd15-cf3f-45cf-8084-d19891f117e4.png)

**Reply by mr-manuel** [reply]
You have to check `com.victronenergy.battery.ttyUSB2`. You can open it with the right arrow on your keyboard.. There you see all data that is seen by the driver.

**Reply by DIVERSDIVERS** [reply]
here
![Bild_2023-04-06_115253275](https://user-images.githubusercontent.com/129951689/230342370-e64dcb63-69ed-40ce-91cb-c46d0d534fe4.png)

**Reply by mr-manuel** [reply]
The data is not read from the Daly BMS. Please open an issue for that. Since I have no Daly BMS I unfortunately cannot help you further.
Please post driver version, Venus OS Version, BMS connection type, BMS make and BMS hard/firmware version (found in the BMS app).

**Reply by DIVERSDIVERS** [reply]
ok i will do thanks for help Manuel.

FYI over the TOP BMS Software all cells could read, with same Cabel
![bms](https://user-images.githubusercontent.com/129951689/230346249-c235b852-6724-4a52-a4a4-20b3401b425c.jpg)

**Reply by DIVERSDIVERS** [reply]
here the cell info, but i do not want to waste your time
![Unbenannt](https://user-images.githubusercontent.com/129951689/230348219-e366764f-5f13-464d-a8ac-71d0c84b7425.JPG)

**Reply by mr-manuel** [reply]
The problem is that probably the current daly battery driver of this script cannot handle yet the latest daly firmware and thus needs to be adjusted.

**Reply by DIVERSDIVERS** [reply]
ok may one day it is done you can tell me, thanks a lot for your help. Should i marked this as solved?

**Reply by mr-manuel** [reply]
Yes, please mark this as solved.

If you open an issue for the cell tradings of Daly BMS then you will be notified, if it's solved.

**Reply by DIVERSDIVERS** [reply]
ok thank you
